src/test/groovy/hudson/plugins/gradle/injection/MavenExtensionDownloadHandlerTest.groovy

```diff
@@ -0,0 +1,90 @@
+package hudson.plugins.gradle.injection
+
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+import spock.lang.Subject
+
+class MavenExtensionDownloadHandlerTest extends Specification {
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder()
+
+    @Subject
+    private final MavenExtensionDownloadHandler mavenExtensionDownloadHandler = new MavenExtensionDownloadHandler()
+
+    def 'extensions are not redonwloaded if config has not changed'() {
+        given:
+        def controllerFolder = tempFolder.newFolder()
+
+        def originalConfig = Mock(InjectionConfig)
+        with(originalConfig) {
+            enabled >> true
+            server >> 'https://scans.gradle.com'
+            mavenExtensionVersion >> '1.22.1'
+            ccudExtensionVersion >> '2.0.1'
+        }
+
+        when:
+        def extensionsDownloaded = mavenExtensionDownloadHandler.ensureExtensionsDownloaded(controllerFolder, originalConfig)
+
+        then:
+        extensionsDownloaded.size() == 2
+        def originalDevelocityDigest = extensionsDownloaded.get(MavenExtension.DEVELOCITY)
+        def originalCcudDigest = extensionsDownloaded.get(MavenExtension.CCUD)
+
+        originalDevelocityDigest != null && originalCcudDigest != null
+
+        when:
+        def sameExtensions = mavenExtensionDownloadHandler.ensureExtensionsDownloaded(controllerFolder, originalConfig)
+
+        then:
+        sameExtensions.size() == 2
+        sameExtensions.get(MavenExtension.DEVELOCITY) == originalDevelocityDigest && sameExtensions.get(MavenExtension.CCUD) == originalCcudDigest
```

Comment:
🤔 how does it test if it's not re-downloaded?

Reply:
Just by verifying that the digest of the one downloaded before hasn't changed.
Or do you mean that it can still download it, just the same version again, which will produce same digest?

Reply:
yes, exactly

Reply:
Added additional check for lastModified